be/src/exec/tablet_sink.cpp

```diff
@@ -571,7 +577,7 @@ void IndexChannel::add_row(Tuple* tuple, int64_t tablet_id) {
         // if this node channel is already failed, this add_row will be skipped
         auto st = channel->add_row(tuple, tablet_id);
         if (!st.ok()) {
-            mark_as_failed(channel, st.get_error_msg(), tablet_id);
+            mark_as_failed(channel.get(), st.get_error_msg(), tablet_id);
```

Comment:
Why not change `mark_as_failed` signature to accept shared_ptr ?

Reply:
I think it is not necessary, and it will cause extra shared ptr copy.

Reply:
But I don't think it's a good practice to mix smart pointers and raw pointers. When we use smart pointers, we already imply that object ownership is managed by smart pointers instead of accessing raw pointers.